html5lib/_utils.py

```diff
@@ -65,6 +70,41 @@ def __getitem__(self, key):
         return dict.get(self, key, self.default)
 
 
+class BoundMethodDispatcher(Mapping):
+    def __init__(self, instance, dispatcher):
+        self.instance = instance
+        self.dispatcher = dispatcher
+        self.cache = {}
+
+    def __getitem__(self, key):
+        if key in self.cache:
+            return self.cache[key]
+
+        try:
+            item = self.dispatcher[key]
+        except KeyError:
+            item = self.default
```

Comment:
It's unclear to me where `self.default` comes from; I have the impression this case is impossible to reach.